### PR TITLE
[formrecognizer] Typing updates

### LINKED ISSUE #26834: [formrecognizer] Improve model typing

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes #26834
Improve typing, enable more typing checks

### CODE REVIEW

File: tools/azure-sdk-tools/ci_tools/environment_exclusions.py [hunk 1]
```diff
@@ -228,6 +228,7 @@
     "azure-eventhub-checkpointstoreblob",
     "azure-eventhub-checkpointstoreblob-aio",
     "azure-eventhub-checkpointstoretable",
+    "azure-identity",
```
Comment (catalinaperalta) [inline tools/azure-sdk-tools/ci_tools/environment_exclusions.py:231]: Something weird happened when merging with upstream here. Will double check that this is currently included on the list.

Comment (catalinaperalta) [reply tools/azure-sdk-tools/ci_tools/environment_exclusions.py]: Fixed

**Comment by catalinaperalta** [comment]
NOTE: I reorganized some models so that they're declared before they're referenced from other models, the git diff may be a bit confusing due to that

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 2]
```diff
@@ -231,70 +232,114 @@ class FormContentType(str, Enum, metaclass=CaseInsensitiveEnumMeta):
...
 
+class TextAppearance:
+    """An object representing the appearance of the text line.
+
+    .. versionadded:: v2.1
+        Support for *to_dict* and *from_dict* methods
+    """
+    style_name: str
+    """The text line style name.
+        Possible values include: "other", "handwriting"."""
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:272]: I noticed when I generated documentation that the `:` did a weird thing and created a `Type` in the docstring. Not sure if you'll see the same thing when you run sphinx but I just removed the `:` and replaced with `possible values are...`

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:272]: Gave the docs a skim and didnt see this causing a problem, however I'll do a more in-depth check, tracking issue: https://github.com/Azure/azure-sdk-for-python/issues/28326

**Pull request merged**